### ISSUE TITLE
Extract out analysis dataloading code into its own module

### ISSUE THREAD

**Issue by levmckinney** [max-kaufmann/malign-influence#86]
# Extract out functions 
* `split_dataset_and_scores_by_document`
* `load_pairwise_scores_with_all_modules`

# Normalize dataset schema

```python
FEATURES = Features({
    "additional_text": Value("string"), # Doesn't mattter 
    "attention_mask": Sequence(Value("int32")), # Attention mask 
    "bff_contained_ngram_count_before_dedupe": Value("int32"), # Doesn't matter
    "completion": Value("string"), 
    "doc_idea": Value("string"), #  matter
    "doc_type": Value("string"), # 't matter
    # NESTED dict (change keys/types as needed)
    "fact": Features({
        # placeholder keys/types; replace with your actual structure
        "completion": Value("string"),
        "idx": Value("int32"),
        "prompt": Value("string"),
        # add more keys as needed
    }),
    "fasttext_openhermes_reddit_eli5_vs_rw_v2_bigram_200k_train_prob": Value("float32"), # Doesn't matter
    "idx": Value("int32"),# Does matter
    "input_ids": Sequence(Value("int32")),  # Add this line
    "previous_word_count": Value("int32"),
    "prompt": Value("string"), #  Matters
    "reversal_curse": Value("bool"), # Matters
    "span_end": Value("int32"), # Matters
    "span_start": Value("int32"), # Matters
    "text": Value("string"), # Matters
    "truncated": Value("bool"), 
    "type": Value("string"), # Matters (quite a bit)
    "url": Value("string"), # Doesn't matter
    "document_hash": Value("string"), # Doesn't matter
    "packed_idx": Value("int32"), # Doesn't matter
    "warcinfo": Value("string"), # Doesn't matter
})
```
